### PR TITLE
Cardano small updates

### PR DESCRIPTION
**Pull request by gabrielKerekes**
This PR contains small refactorings and updates for the Cardano app.
- address validation has so far been coupled with address derivation - this PR splits out address validation - https://github.com/vacuumlabs/trezor-firmware/issues/59
- we also decided to introduce validation to check that no extra fields are set in address parameters
- transaction outputs can no longer contain both `address` and `address_parameters` - previously if both were set were set `address` would simply be ignored and `address_parameters` would be used
- we decided to re-introduce the MAX_TX_OUTPUT_SIZE limitation so that Trezor wouldn't be able to create an output, which it later wouldn't be able to spend because of memory constraints - https://github.com/vacuumlabs/trezor-firmware/issues/58

### CODE REVIEW

File: core/src/apps/cardano/address.py [hunk 1]
```diff
@@ -34,6 +41,73 @@
...
+
+
+def _validate_address_parameters_structure(
+    parameters: CardanoAddressParametersType,
+) -> None:
+    address_n_staking = parameters.address_n_staking
+    staking_key_hash = parameters.staking_key_hash
+    certificate_pointer = parameters.certificate_pointer
+
+    fields_to_be_empty: List[
```
[the remark this answers is not in the document]
Comment (matejcik) [reply core/src/apps/cardano/address.py:75]: also your previous revisions was using tuples, ISTM declaring `fields_to_be_empty: tuple` beforehand would fix that?

**Review by matejcik — CHANGES_REQUESTED** [review]
very nice
one point about typing annotations, otherwise LGTM

**Comment by matejcik** [comment]
also please add a changelog per https://docs.trezor.io/trezor-firmware/misc/changelog.html

**Comment by matejcik** [comment]
it seems some of your address test vectors are wrong now: https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/1318118622

File: core/src/apps/cardano/address.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gabrielKerekes) [reply core/src/apps/cardano/address.py]: I reverted back to using tuples. The only problem I had with using them was that the inner type would be complicated. But if it's ok to use just `tuple` without an inner type then that's great.

**Comment by gabrielKerekes** [comment]
> also please add a changelog per https://docs.trezor.io/trezor-firmware/misc/changelog.html

Changelogs added - https://github.com/trezor/trezor-firmware/pull/1606/commits/0010bbc58277a4c581af948c416cad136b820747. I am not sure if two entries are desirable or just one would be enough similar to `Cardano: refactor and improve address validation and reintroduce maximum transaction output size`. Or perhaps there should be one entry for each point in the PR description.

> it seems some of your address test vectors are wrong now: https://gitlab.com/satoshilabs/trezor/trezor-firmware/-/jobs/1318118622

Indeed. Luckily nothing got broken, it really was just an invalid test vector. Fixed in https://github.com/trezor/trezor-firmware/pull/1606/commits/7c573c98126382883d31639f500af06c70e8d913.

**Comment by matejcik** [comment]
> I am not sure if two entries are desirable

these entries seem appropriate :+1: 

in the meantime we changed import paths so you got some conflicts, please squash the fixups and rebase on top of master

**Comment by gabrielKerekes** [comment]
> please squash the fixups and rebase on top of master

Done.

**Review by matejcik — APPROVED** [review]

**Comment by matejcik** [comment]
and done, thanks!

**Pull request merged**